onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_api_helper.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include "core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.h"
+
+#ifdef __ANDROID__
+#include <android/api-level.h>
+#endif
+
+struct NnApi;
+
+namespace onnxruntime {
+namespace nnapi {
+
+class ModelBuilder;
+
+enum class TargetDeviceOption : int8_t {
+  ALL_DEVICES,  // use all available target devices
+
+  /* TODO support these options
+  PREFERRED_DEVICES,  // Use one or more preferred devices (must be given)
+  EXCLUDED_DEVICES,   // Exclude one or more devices (must be given)
+   */
+
+  CPU_DISABLED,  // use all available target devices except CPU
+  CPU_ONLY,      // use CPU only
+};
+
+int32_t GetNNAPIRuntimeFeatureLevel(const ::NnApi* nnapi);
+
+int32_t NNAPIGetTargetFeatureLevel(const ::NnApi* nnapi, const std::vector<ANeuralNetworksDevice*>& device_handles);
```

Comment:
- naming suggestions
  - "NNAPIGetTargetFeatureLevel" -> "GetNNAPITargetFeatureLevel"
    - or "GetNNAPIEffectiveFeatureLevel", where "effective" means the feature level that we should actually use (e.g., feature level supported by all devices)
  - "NNAPIGetTargetFeatureLevelWithDeviceTag" -> "GetNNAPITargetFeatureLevelFromTargetDeviceOption" or "GetNNAPIEffectiveFeatureLevelFromTargetDeviceOption"
- perhaps we don't need `NNAPIGetTargetFeatureLevel(const ModelBuilder& model_builder)`

Reply:
Here I didn't handle failure of GetTargetDevices. -1 will be returned for  it will refuse all ops for NNAPI.